internal/olm/operator/registry/index/registry_pod.go

```diff
@@ -77,15 +80,14 @@ type RegistryPod struct {
 	// GRPCPort is the container grpc port
 	GRPCPort int32
 
-	// client refers to a controller runtime client
-	client client.Client
-
 	// pod represents a kubernetes *corev1.pod that will be created on a cluster using an index image
 	pod *corev1.Pod
+
+	cfg *operator.Configuration
 }
 
 // NewRegistryPod initializes the RegistryPod struct and sets defaults for empty fields
-func NewRegistryPod(client client.Client, dbPath, bundleImage, namespace string) (*RegistryPod, error) {
+func NewRegistryPod(cfg *operator.Configuration, dbPath, bundleImage, namespace string) (*RegistryPod, error) {
```

Comment:
Namespace is included in `cfg` so you don't need a parameter or field for one.
```suggestion
func NewRegistryPod(cfg *operator.Configuration, dbPath, bundleImage string) (*RegistryPod, error) {
```